test/test_matrix4x4.py

```diff
@@ -0,0 +1,241 @@
+    matrix = zivid.Matrix4x4()
+
+    for i in range(4):
+        for j in range(4):
+            matrix[i, j] = sample_2d_list()[i][j]
+            assert matrix[i, j] == sample_2d_list()[i][j]
+
+    assert_all_equal_2d(matrix, sample_2d_list())
+
+    for i in range(-4, 0):
+        for j in range(-4, 0):
+            matrix[i, j] = sample_2d_list()[i][j]
+            assert matrix[i, j] == sample_2d_list()[i][j]
+
+    assert_all_equal_2d(matrix, sample_2d_list())
+
+    with pytest.raises(TypeError):
+        matrix[0] = 0
+
+    with pytest.raises(TypeError):
+        matrix[0, 0, 0] = 0
+
+    with pytest.raises(TypeError):
+        matrix[1.4, 1.0] = 0
+
+    with pytest.raises(TypeError):
+        matrix["0", "0"] = 0
+
+    with pytest.raises(TypeError):
+        matrix[0, 0] = "42"
+
+    with pytest.raises(TypeError):
+        matrix[0, 0] = 10 ** 1000
+
+    with pytest.raises(IndexError):
+        matrix[1000, 0] = 0
+
+    with pytest.raises(IndexError):
+        matrix[0, 1000] = 0
+
+    with pytest.raises(IndexError):
+        matrix[0, -1000] = 0
+
+
+def test_rows_cols():
+    assert zivid.Matrix4x4.rows == 4
+    assert zivid.Matrix4x4.cols == 4
+    assert zivid.Matrix4x4().rows == 4
+    assert zivid.Matrix4x4().cols == 4
+
+    with pytest.raises(AttributeError):
+        zivid.Matrix4x4.rows = 3
+
+    with pytest.raises(AttributeError):
+        zivid.Matrix4x4.cols = 3
+
+
+def test_inverse():
+    def invertible_matrix():
+        return [
+            [1, 1, 1, -1],
+            [1, 1, -1, 1],
+            [1, -1, 1, 1],
+            [-1, 1, 1, 1],
+        ]
+
+    def non_invertible_matrix():
+        return [1] * 16
+
+    matrix = zivid.Matrix4x4(invertible_matrix())
+
+    assert_all_equal_2d(
+        matrix.inverse(),
+        [
+            [0.25, 0.25, 0.25, -0.25],
+            [0.25, 0.25, -0.25, 0.25],
+            [0.25, -0.25, 0.25, 0.25],
+            [-0.25, 0.25, 0.25, 0.25],
+        ],
+    )
+
+    assert_all_equal_2d(matrix, invertible_matrix())
+
+    matrix = zivid.Matrix4x4(non_invertible_matrix())
+
+    with pytest.raises(RuntimeError):
+        matrix.inverse()
+
+    assert_all_equal_flat(matrix, non_invertible_matrix())
+
+
+def test_buffer_protocol():
+    numpy_array = numpy.array(zivid.Matrix4x4(sample_2d_list()))
+    assert (numpy_array == sample_2d_list()).all()
+
+
+def test_to_string():
+    matrix = zivid.Matrix4x4(sample_2d_list())
+
+    assert str(matrix) == (
+        "[ [ 0.000000,  1.000000,  2.000000,  3.000000], \n"
+        "  [ 4.000000,  5.000000,  6.000000,  7.000000], \n"
+        "  [ 8.000000,  9.000000,  10.000000,  11.000000], \n"
+        "  [ 12.000000,  13.000000,  14.000000,  15.000000] ]"
+    )
+
+
+def test_save():
+    with tempfile.TemporaryDirectory() as tmpdir, zivid.Application() as _:
+        file = os.path.join(tmpdir, "matrix_saved.yml")
+        matrix = zivid.Matrix4x4(
+            [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11], [12, 13, 14, 15]]
```

Comment:
Could you change some of these values to be floating points? This way we also check that it does handle floating point values.